hack/log.sh

```diff
@@ -0,0 +1,215 @@
+#!/bin/bash
+#
+# This library holds all of the logging functions for OpenShift bash scripts.
+
+# os::log::install_cleanup installs os::log::clean_up_logger as a trap on interrupts, termination, and exit
+#
+# Globals:
+#  None
+# Arguments:
+#  None
+# Returns:
+#  None
+function os::log::install_cleanup() {
+    trap "os::log::clean_up_logger; $(trap -p SIGINT)" SIGINT
+    trap "os::log::clean_up_logger; $(trap -p SIGTERM)" SIGTERM
+    trap "os::log::clean_up_logger; $(trap -p EXIT)" EXIT
+}
+
+# os::log::clean_up_logger should be trapped so that it can stop the logging utility once the script that
+# installed it is finished.
+# This function stops logging and generates plots of data for easy consumption.
+#
+# Globals:
+#  - LOG_DIR
+#  - LOGGER_PID
+#  - SAR_LOGFILE
+# Arguments:
+#  None
+# Returns:  
+#  None
+function os::log::clean_up_logger() {
+    local return_code
+    return_code=$?
+
+    if jobs -pr | grep -q "${LOGGER_PID}"; then
```

Comment:
*even though* I am doing this entire block on this `if` statement, which checks if the PID is in the list of running child processes of this PID, I *still* sometimes get:
```
/home/stevekuznetsov/go/src/github.com/openshift/origin/hack/log.sh: line 36: kill: (7094) - No such process
```

I have no idea how that is possible.